### ISSUE TITLE
Possible Memory Leak in scipy.spatial.distance.pdist and scipy.spatial.distance.cdist

### ISSUE THREAD

**Issue by MariusGachter** [scipy/scipy#14502]
My issue is about a possible memory leak in scipy.spatial.distance.pdist and scipy.spatial.distance.cdist

#### Reproducing code example:

```
import numpy as np
from scipy.spatial.distance import pdist
X = np.random.uniform(0, 100, size=10000).reshape(-1, 10)
for _ in range(10000):
    test = pdist(X/0.01, metric='sqeuclidean')
    
```

#### Error Description:
On my Mac OSX 10.14.6 the memory consumption of the snipped above quickly rises into the Gigabytes (after a few seconds). If I remove the division from the pdist argument (see below), the problem vanishes. (I don't think this is intended behavior, as it also leads to excessive memory consumption in the Gaussian Process Regression implementation of Sklearn, where they have a similar line in the implementation of the RBF kernel)

The code, which does not produce this problem (division removed):
```
import numpy as np
from scipy.spatial.distance import pdist
X = np.random.uniform(0, 100, size=10000).reshape(-1, 10)
for _ in range(10000):
    test = pdist(X, metric='sqeuclidean')
```

#### Scipy/Numpy/Python version information:

```
1.7.0 1.21.0 sys.version_info(major=3, minor=8, micro=5, releaselevel='final', serial=0)
```

**Comment by peterbell10** [comment]
This is a duplicate of  #14382 and should be fixed in the next minor release.

**Comment by tylerjereddy** [comment]
Fixed by https://github.com/scipy/scipy/pull/14384 ; I wonder if I need to think about getting that out sooner than later.

**Comment by MariusGachter** [comment]
> Fixed by #14384 ; I wonder if I need to think about getting that out sooner than later.

Thanks a lot! By installing from source, I indeed managed to verify, that the problem seems to be solved. 
I personally would appreciate being able to install this new version via pip, as I need to also run my code on a cluster, where I doubt that I have enough privileges to build from source.
